workers/ohsome_quality_analyst/ohsome/client.py

```diff
@@ -69,3 +61,74 @@ async def get_latest_ohsome_timestamp():
     timestamp_str = str(resp.json()["extractRegion"]["temporalExtent"]["toTimestamp"])
     timestamp = datetime.datetime.strptime(timestamp_str, "%Y-%m-%dT%H:%MZ")
     return timestamp
+
+
+def build_url(
+    layer,
+    endpoint: Optional[str] = None,
+    ratio: bool = False,
+) -> str:
+    """Build endpoint URL of ohsome API."""
+    ohsome_api = OHSOME_API.rstrip("/")
+    if endpoint is None:
+        endpoint = layer.endpoint
+    url = ohsome_api + "/" + endpoint.rstrip("/")
+    if ratio:
+        return url + "/" + "ratio"
+    return url
+
+
+# TODO: Add documentation and tests on time string format.
+def build_data_dict(
+    layer,
+    bpolys: str,
+    time: Optional[str] = None,
+    ratio: bool = False,
+) -> dict:
+    """Build data dictionary for ohsome API query."""
+    data = {"bpolys": bpolys, "filter": layer.filter}
+    if ratio:
+        if layer.ratio_filter is None:
+            raise ValueError(
+                "Layer '{0}' has not 'ratio_filter' definied.".format(layer.name)
+            )
+        else:
+            data["filter2"] = layer.ratio_filter
+    if time is not None:
+        data["time"] = time
+    return data
+
+
+def check_iso_time(time: str) -> bool:
```

Comment:
I started to implement this function to validate the time parameter before it is used as query parameter with the ohsome API. But since the ISO-8601 conform timestring(s) can be quite versatile it can only be tested rudimentary with the built-in stdlib. An alternative is to use a library such as dateutils to validate the timestring. Another one is to forgo it and pass it to the ohsome API in the hope to catch invalid timestring through the response of the ohsome API. 

@joker234 I would be happy to hear your thoughts on that.

Reply:
I like the idea to catch invalid timestrings through the response of ohsome API.

While playing around with the ohsome API I found out two different response related to wrong use of the time parameter:

```
{
  "timestamp": "2021-05-31T07:33:31.206528",
  "status": 400,
  "message": "You need to give at least two timestamps or a time interval for this resource.",
  "requestUrl": "https://api.ohsome.org/v1/contributions/count?bboxes=8.67%2C49.39%2C8.71%2C49.42&filter=type%3Away%20and%20natural%3D*&format=json&time=20140101"
}

{
  "timestamp": "2021-05-31T07:36:40.661092",
  "status": 400,
  "message": "The provided time parameter is not ISO-8601 conform.",
  "requestUrl": "https://api.ohsome.org/v1/contributions/count?bboxes=8.67%2C49.39%2C8.71%2C49.42&filter=type%3Away%20and%20natural%3D*&format=json&time=20140101%2C2015010"
}
```

@tyrasd are there more cases that should be covered?

Reply:
I removed functions for testing the time parameter. 
The code in line 43 will raise an exception if the ohsome response will return status code 400 when the time parameter is invalid.
```python
    try:
        resp.raise_for_status()  # Raise for response status codes 4xx and 5xx
    except httpx.HTTPStatusError as error:
        raise OhsomeApiError(
            "Querying the ohsome API failed!" + error.response.json()["message"]
        ) from error
```

Which will lead to following Error:

```
Traceback (most recent call last):
  File "/home/matthias/work/oqt/test.py", line 7, in <module>
    resp.raise_for_status()  # Raise for response status codes 4xx and 5xx
  File "/home/matthias/.cache/pypoetry/virtualenvs/ohsome-quality-analyst-JhxvPu0L-py3.9/lib/python3.9/site-packages/httpx/_models.py", line 1105, in raise_for_status
    raise HTTPStatusError(message, request=request, response=self)
httpx.HTTPStatusError: 400 Client Error:  for url: https://api.ohsome.org/v1/contributions/count?bboxes=8.67%2C49.39%2C8.71%2C49.42&filter=type%3Away%20and%20natural%3D*&format=json&time=20140101%2C2015010
For more information check: https://httpstatuses.com/400

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/matthias/work/oqt/test.py", line 10, in <module>
    raise OhsomeApiError("Querying the ohsome API failed! " + _) from error
ohsome_quality_analyst.utils.exceptions.OhsomeApiError: Querying the ohsome API failed! The provided time parameter is not ISO-8601 conform.
```